### ISSUE TITLE
Mono + dotnet/runtime Migration

### ISSUE THREAD

**Issue by steveisok** [dotnet/runtime#1018]
These are the high level sets of tasks / priorities leading up to and after parts of mono are migrated into dotnet/runtime.  

## Before Migration
- [x] Determine steps needed to enable dotnet/runtime testing for mono
- [x] Enable PR Sync Bot and test 

## Migration
- [x] Execute mono consolidation script (requires mono merge freeze)
- [x] Merge PR into dotnet/runtime: https://github.com/dotnet/runtime/pull/1912
- [x] Enable PR Sync Bot 

### Priority I (Near completion)
- [x] Build MonoVM at the same time as CoreCLR
- [x] Enable LLVM support + binary packages: https://github.com/dotnet/runtime/pull/2140
- [x] Adjust, if necessary, helper targets like `run-sample` `run-benchmark`: https://github.com/dotnet/runtime/pull/2393
- [x] Migrate ios build targets: https://github.com/dotnet/runtime/pull/33292
- [x] Enable CI for ios: https://github.com/dotnet/runtime/issues/33688
- [x] Migrate android build targets: https://github.com/dotnet/runtime/issues/33687 and https://github.com/dotnet/runtime/pull/33881
- [x] Enable CI for android: https://github.com/dotnet/runtime/issues/33689
- [x] Runtime packs for ios and android: https://github.com/dotnet/runtime/issues/33690

### Priority II
- [x] Migrate tvOS build targets: https://github.com/dotnet/runtime/issues/34461
- [x] Default to LLVM + AOT for device release build: https://github.com/dotnet/runtime/issues/34693
- [x] Migrate ios test targets : https://github.com/dotnet/runtime/issues/33753
- [x] Get ios test targets running on devices / simulators
- [x] Migrate android test targets
- [x] Get android test targets running on devices / simulators
- [x] Migrate wasm build targets: https://github.com/dotnet/runtime/issues/33818
- [x] Migrate wasm test targets
- [ ] Meta-package for LLVM dependencies
- [ ] Run `mono/tests` tests and `mono/mini` tests on MonoVM on CI
- [ ] Run MonoVM's soft debugger tests on CI (this will need some coordination with https://github.com/mono/debugger-libs)

**Comment by jkotas** [comment]
What is PR Sync Bot?

**Comment by danmosemsft** [comment]
Are these worth calling out specifically

* Add build configurations for iOS and Android
* Porting of code into System.Native
* Porting of code into libraries, changing #ifdef to partial classes in many cases
* Add build configuration for Windows desktop (?) and CI (?)

Post migration
* Size measurements, trimming as necessary, infrastructure to protect size.

**Comment by marek-safar** [comment]
@danmosemsft The goal is to get the build going first. Any implementation changes will come once the infrastructure is in place and working.

**Comment by lambdageek** [comment]
A few more post migration tasks:
- run `mono/tests` tests and `mono/mini` tests on MonoVM on CI
- run MonoVM's soft debugger tests on CI (this will some coordination with https://github.com/mono/debugger-libs)

**Comment by akoeplinger** [comment]
I just opened the PR that adds Mono to the repo: https://github.com/dotnet/runtime/pull/1912 :)

**Comment by borgdylan** [comment]
What will the story be for those running mono on Linux? Would there still be a way to build it and install it via makefiles? Will there be a similar story for .NET 5 as a replacement?

**Comment by akoeplinger** [comment]
@borgdylan Mono as you know it will stay around as an option, but for .NET 5 the packaging/distribution is a bit different.

**Comment by shahid-pk** [comment]
is this just port of monovm, which will use .net core gc and JIT. Or is the entire mono runtime is getting ported here ?
Interested because mono runtime runs in many places where .net core does not and .net core's gc is way better than mono's gc.

**Comment by akoeplinger** [comment]
It is the entire Mono runtime including Mono's GC (sgen).

**Comment by tibel** [comment]
Is there a plan to port Mono to use the .NET Core GC?
I would prefer having one garbage collector for only in terms of testing and maintenance.

**Comment by jkotas** [comment]
> Is there a plan to port Mono to use the .NET Core GC?

Xamarin and Blazor are the primary workloads for Mono. CoreCLR GC (as is) is likely a worse option for these workloads than the current Mono GC. We may experiment with it, but the Mono GC is here to stay until we have a data that confirm the .NET Core GC would be a better choice for the primary Mono workloads.

**Comment by richlander** [comment]
I don't see anything in that list for publishing runtime packs. Is that missing? If so, it seems like it should be added. W/o that, there isn't a user experience.

**Comment by jaredpar** [comment]
When doing the planning for migrating the mono runtime into the repository there was no cleanly defined user experience. Given that we decided to cut the runtime packs and publishing because there wasn't a clear goal for us to plan around. Instead we decided to separate migrating mono into the runtime and then work on publishing, official builds and packaging once we had a defined scenario to plan that around.

Is that available now for us to look at?
